### PR TITLE
Add BYOVPC Testing Support

### PR DESCRIPTION
**Pull request by gene-redpanda**
Adds support for tests for BYOVPC and fixes a bug related to increasing partitions on a topic. 

The test is not currently active as it has run into an edge case on the cloud side that prevents clusters it creates from being deleted.

### CODE REVIEW

File: redpanda/tests/acceptance_test.go [hunk 1]
```diff
@@ -22,7 +23,7 @@ const (
 	gcpDedicatedClusterFile   = "../../examples/cluster/gcp/main.tf"
 	serverlessClusterFile     = "../../examples/cluster/serverless/main.tf"
 	awsByocClusterFile        = "../../examples/byoc/aws/main.tf"
-	awsByocVpcClusterFile     = "../../examples/byovpc/aws/main.tf"
+	awsByocVpcClusterFile     = "../../infra/byovpc/aws/main.tf"
```
Comment (andrewstucki) [inline redpanda/tests/acceptance_test.go:26]: So, overall this seems fine to me, but TBH I'm not thrilled about the top-level `infra` directory. Is there a reason not to put this somewhere like `redpanda/tests/infra`?

File: redpanda/validators/zone_id.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+		return
+	}
+
+	// If cloud_provider is not aws or unknown, skip validation
+	if cloudProvider.IsUnknown() || cloudProvider.ValueString() != "aws" {
+		return
+	}
+
+	// AWS Zone ID regex pattern (e.g., use1-az1)
+	zoneIDPattern := regexp.MustCompile(`^[a-z]{2,5}\d-az\d$`)
```
Comment (andrewstucki) [inline redpanda/validators/zone_id.go:62]: nit: this should probably be pulled out into a top-level variable rather than re-compiling the regex every validation call.

**Review by andrewstucki — APPROVED** [review]
So if I'm parsing all of this correctly it seems like we have a few changes here:

1. Adding in a validator for AWS zone ids
2. Increasing the timeout for cluster resource creation
3. moving around the schemas for some of the resources (I'm assuming for consistency in file convention naming)
4. ensuring partition counts only increase
5. adding in the byovpc `infra` folder in order to split out network creation from the cluster

If I missed anything, let me know, but aside from groking this correctly, the only real question I have here is about the directory structure here that seems a bit wonky: `infra` which is only clearly used in tests being a top-level folder.

Approving regardless as the actual code changes seem fine, but if you're not adamantly against it, I'd love to move that `infra` folder.

**Comment by gene-redpanda** [comment]
Comments resolved!
